File: 2019/02/python/main.py
#!/usr/bin/env python3

from intcode import Intcode
from sys import exit

def run() -> str:
  f = open('../input.txt', 'r')
  codes = [int(code) for code in f.read().strip().split(',')]
  f.close()

  expectation = 19690720
  
  print("Calculating", end = "")
  for noun in range(100):
    print(".", end = "")
    for verb in range(100):
      intcode = Intcode(codes)
      result = intcode.run(noun, verb)
      if result == expectation:
        return "\nnoun({}) * 100 + verb({}) = {}".format(noun, verb, noun * 100 + verb)

if __name__ == "__main__":
  print(run())

File: 2019/02/ruby/main.rb
#!/usr/bin/env ruby

require_relative 'intcode'

input = File.read('../input.txt')
codes = input.chomp.split(',').map(&:to_i)

expectation = 19690720

print "Figuring out"
(0..99).each do |noun|
  print "."
  (0..99).each do |verb|
    intcode = Intcode.new(codes).run(noun, verb)
    if intcode == expectation
      puts "\nnoun(#{noun}) * 100 + verb(#{verb}) = #{noun * 100 + verb}"
      exit
    end
  end
end
